stout/event-loop.h

```diff
@@ -132,6 +132,51 @@ class EventLoop : public Scheduler {
     uv_buf_t buffer_ = {};
   };
 
+  class Signal {
+   public:
+    Signal(EventLoop& loop) {
+      handle_ = new uv_signal_t();
+      CHECK_EQ(
+          uv_signal_init(loop, handle_),
+          0);
+    }
+
+    Signal(const Signal& that) = delete;
+
+    Signal(Signal&& that) {
+      handle_ = that.handle_;
+      that.handle_ = nullptr;
+    }
+
+    ~Signal() {
+      if (handle_) {
+        if (uv_is_active(reinterpret_cast<uv_handle_t*>(handle_))) {
+          uv_signal_stop(handle_);
+        }
+
+        uv_close(
+            reinterpret_cast<uv_handle_t*>(handle_),
+            EventLoop::CloseCallback);
+      }
+    }
+
+    // Adaptors to libuv functions.
+    uv_signal_t* handle() {
+      CHECK(handle_);
```

Comment:
Here too (and below) I'd like to be explicit about the type:
```suggestion
      CHECK_NOTNULL(handle_);
```